pkg/stream/producer_unconfirmed.go

```diff
@@ -118,3 +130,12 @@ func (u *unConfirmed) size() int {
 	defer u.mutexMessageMap.Unlock()
 	return len(u.messages)
 }
+
+func (u *unConfirmed) maybeUnLock() {
+	if len(u.messages) < u.maxSize {
+		logs.LogDebug("unConfirmed size: %d back to normal, producer unblocked", u.maxSize)
+		u.blockSignal.L.Lock()
+		u.blockSignal.Signal()
```

Comment:
```suggestion
		u.blockSignal.Broadcast()
```
I would use broadcast, because in case of concurrent call to  `BatchSend` or `Send` (in this case with a message bigger than maxFrameSize) this will unlock only one call, while the other will stay blocked.
On the max size the control will be weaker, but it would avoid blocking calls.